remote-api/src/main/java/org/alfresco/rest/api/impl/CategoriesImpl.java

```diff
@@ -152,6 +153,23 @@ private NodeRef createCategoryNodeRef(NodeRef parentNodeRef, Category c)
         return categoryService.createCategory(parentNodeRef, c.getName());
     }
 
+    @Override
+    public void deleteCategoryById(String id, Parameters params)
+    {
+        if (!authorityService.hasAdminAuthority())
+        {
+            throw new PermissionDeniedException(NO_PERMISSION_TO_DELETE_A_CATEGORY);
+        }
```

Comment:
I think Krys rewrote this class a bit too - there's a helper method to check if the user is an admin.